### ISSUE TITLE
Text alignment and side-by-side editboxes

### ISSUE THREAD

**Issue by sburkett** [sirinsidiator/ESO-LibAddonMenu#124]
Not sure if this is an issue, a suggestion, or a plea for help :) I did some searching but came up empty, so apologies if this is obvious and I've missed something right in front of me.

What I'm trying to do is something like this:

![image](https://user-images.githubusercontent.com/2219483/103800649-a9054a00-501a-11eb-813a-f00f1b46dec3.png)

So, basically, a series of side-by-side editboxes, with 2 column headers at the top.

Ideally, I'd like the fields to be aligned with the column headers, but they appear offset as you can see. I see where this is a native `SetHorizontalAlignment()` method in ESO, but trying that manually didn't seem to work.

Is there some way to align those editboxes to the left rather than the right?

Any thoughts?

**Comment by Baertram** [comment]
The gap exists because you are using the width="half" tag at the editboxes. With width="half" the gap always exists because of the name shown at the top left edge of the edit control. Afaik this was done to be able to show the name/title for the total of 50% of the lam panel's width. Else the title/name would be abbreviated like "Column ..." very early if it does not fit into the space between the left lam panel border and the editbox (50% = "half" of the lam panel).
Maybe it's even a technical issue with the label used there.
So with the given lam editbox you cannot change this gap afaik ,and you cannot align them to the left, no.

The conrols use anchors (to eah other, to the lam panel, to the gui elements/other controls/parent controls etc.). If you want to change the anchors you need to use control:ClearAnchors() at first and after that you can use other control functions that change the alignment + reapply SetAnchor() to anchor them properly to other controls again.

You could build your own anchored editboxes like you want them to be within the LAM custom control widget.
It got a created funciton callback and a minHeight attribute to preset the custom control's height. You are able to add controls to that custom contol and anchor them the way you want them to appear. Maybe there even is a way to re-use the LAM editbox controls there.

**Comment by Baertram** [comment]
Oh and maybe your usecase could be handled differently? Have a look at the library LibShifterBox. It got 2 list controls where you are able to move items from left to right. Maybe it helps with your editbox list somehow?
You can add this to a LAM customcontrol e.g.

**Comment by sburkett** [comment]
There doesn't seem to be much documentation on the newer custom control feature. Is the custom control basically a "container" for other controls? If so, how do you place other controls in it? Is it just a matter of using the custom anchoring functions to put everything relative to this invisible/custom container?

**Comment by Baertram** [comment]
Yep, exactly. It's just a container control where you anchor your other controls into, just like you need them.
Could look like this e.g
https://i.imgur.com/nhXL1la.png
Yellow rectangle: LAM custom control
Orange rectangle: LibShifterBox control added to the custom control via the "createFunc" callback
